packages/next/src/server/dev/hot-reloader.ts

```diff
@@ -413,6 +419,55 @@ export default class HotReloader {
       )
   }
 
+  private async getVersionInfo(span: Span) {
+    const versionInfoSpan = span.traceChild('get-version-info')
+    return versionInfoSpan.traceAsyncFn<VersionInfo>(async () => {
+      try {
+        const installed = require('next/package.json').version
+        const installedParsed = semver.parse(installed)
+
+        const res = await fetch(
+          'https://registry.npmjs.org/-/package/next/dist-tags'
+        )
+
+        if (!res.ok) return { installed, staleness: 'unknown' }
+
+        const tags = await res.json()
+        const latest = semver.parse(tags.latest)
+        const canary = semver.parse(tags.canary)
+
+        if (installedParsed && latest && canary) {
+          if (installedParsed.major < latest.major) {
+            // Old major version
+            return { staleness: 'outdated', expected: latest.raw, installed }
+          } else if (
+            installedParsed.prerelease[0] === 'canary' &&
+            semver.lt(installedParsed, canary)
+          ) {
+            // Matching major, but old canary
+            return { staleness: 'outdated', expected: canary.raw, installed }
```

Comment:
Perhaps we should change these cases to

- fresh
- stale-patch
- stale-minor
- stale-major
- unknown

Then we can adjust the text the user sees without changing the parsing logic in the future